server/chat.ts

```diff
@@ -1049,10 +1049,9 @@ export class CommandContext extends MessageContext {
 			this.errorReply(this.tr("Your message contained banned words in this room."));
 			return null;
 		}
-
 		const gameFilter = this.checkGameFilter();
-		if (gameFilter && !user.can('bypassall')) {
-			this.errorReply(gameFilter);
+		if (gameFilter && (this.room?.battle || room?.battle || !user.can('bypassall'))) {
+			this.errorReply(gameFilter.trim());
```

Comment:
Oh, I didn't even realize how weird this workaround is. For future reference, you should use a sentinel value like `true` rather than this weirdness (which will send an empty error for no reason).

And anyway, all this is moot because for lag reasons it's much better to just let this one through, and ignore the one that comes from the battle process.